### PR TITLE
[DC-2055] Replace gcs_utils.copy_object() calls with StorageClient.bucket.copy_blob()

### PR DESCRIPTION
**Pull request by lledezma**

### CODE REVIEW

File: tests/unit_tests/data_steward/validation/main_test.py [hunk 1]
```diff
@@ -476,55 +471,58 @@ def test_copy_files_ignore_dir(self, mock_check_cron, mock_hpo_bucket,
...
         self.assertEqual(result, expected)
-        self.assertTrue(mock_check_cron.called)
-        self.assertTrue(mock_hpo_bucket.called)
-        self.assertTrue(mock_drc_bucket.called)
-        self.assertTrue(mock_bucket_list.called)
+        mock_check_cron.assert_called()
+        mock_bucket_list.assert_called()
+
+        mock_client.get_hpo_bucket.assert_called()
+        mock_client.get_drc_bucket.assert_called()
```
Comment (lrwb-aou) [inline tests/unit_tests/data_steward/validation/main_test.py:498]: 👍

**Review by lrwb-aou — APPROVED** [review]
I've added a few optional suggestions.  Let me know what you think.

It looks good.

File: tests/unit_tests/data_steward/validation/main_test.py [hunk 2]
```diff
@@ -476,55 +471,58 @@ def test_copy_files_ignore_dir(self, mock_check_cron, mock_hpo_bucket,
...
-        # that should be used when we upgrade instead of this
+
         for call in unexpected_calls:
             try:
-                mock_copy.assert_has_calls([call], any_order=True)
+                mock_source_bucket.copy_blob.assert_has_calls([call],
+                                                              any_order=False)
             except AssertionError:
                 pass
             else:
```
[the remark this answers is not in the document]
Comment (lledezma) [reply tests/unit_tests/data_steward/validation/main_test.py:528]: It works. Thanks!

**Review by lrwb-aou — APPROVED** [review]
It all looks good!

**Pull request merged**